posts/hello-world-machine.md

```diff
@@ -0,0 +1,252 @@
+
+```python
+def hello() -> str:
+    ascii = [72, 101, 108, 108, 111, 44, 32, 87, 111, 114, 108, 100, 33]
+    return ''.join(chr(value) for value in ascii)
+```
+
+#### The Dictionary
+
+You watched the news and saw a snippet about a man being rescued using S.O.S signal. "Maybe the machine should be able to handle morse code as well!" - you thought, but you are not sure why. Its only job is to return the greeting message...
+
+```python
+def hello(morse) -> str:
+    morse_code = {
+        "...."  : "H",
+        "."     : "e",
+        ".-.."  : "l",
+        "---"   : "o",
+        "--..--": ",",
+        "/"     : " ",
+        ".--"   : "W",
+        ".-."   : "r",
+        "-.."   : "d",
+        "-.-.--": "!"
+         
+    }
+    return "".join(morse_code[value] for value in morse.split())
+
+print(hello(".... . .-.. .-.. --- --..-- / .-- --- .-. .-.. -.. -.-.--"))
+```
+
+#### The Functional
+
+With your current machine are prone to changes and damage, you decide to build a smaller but an immutable and indestructible machine that do the exact same thing. And whenever the regular machine not working or acting up, it will activate the smaller one instead. Hang on, why don't we just use the smaller on from now on..
+
+```python
+def hi() -> str:
+	return "Hello, World!"
+
+def hello() -> str:
+    return hi()
+```
+
+#### The Permutation
+
+A strong thunderstorm went by and your poor unlucky machine got struck by lightning, it still power up but now it is acting weird.  All the characters to assemble the greeting message is still there but they are not in order, and the damaged to the logic board mean the machine can't get the correct order for the message anymore. 
+And just your luck, the incredible reliable and *indestructible* smaller one you built earlier was in fact stealable (Yes, it's gone!). So you decide to just let your good ol' buddy to try every possible ways to scramble these letters into the correct message. After all, it is only 13 characters and the machine has a lot of memory that you recently downloaded, this should be quick...
+
+```python
+from itertools import permutations
+
+def hello() -> str:
+    chars = ['H', 'e', 'l', 'l', 'o', ',', ' ', 'W', 'o', 'r', 'l', 'd', '!']
+
+    all_permutations = permutations(chars)
+    for perm in all_permutations:
+        if ("".join(perm) == "Hello, World!"):
+            return ''.join(perm)
+```
+
+#### The Search
+
+You decided to give an update to the machine, make it learn new language so it can greet in different languages. But its ONLY job is to say "Hello, World!" in English, so in its confusion it decides to download a bazillion ways of saying "Hello, World!", each with slightly different capitalized characters and placement. Now it confused and not sure which one to pick, well, at least these are all in some sort of order. You have to tell machine the correct way to phrase it, so it can find the phrase and be normal again(maybe)
```

Comment:
```suggestion
You decided to update the machine to allow it to greet in different languages.
But its ONLY job is to say "Hello, World!" in English, so in its confusion it decides to download a bazillion ways of saying "Hello, World!", each with slightly different capitalized characters and placement.
That left you confused and not sure which one to pick, but at least they are all in some sort of order.
You have to tell the machine the correct way to phrase it, so it can find the phrase and be normal again(maybe).
```